precompiles/proxy/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+
+		Ok(succeed([]))
+	}
+
+	/// Unregister a proxy account for the sender.
+	/// The dispatch origin for this call must be Signed.
+	///
+	/// Parameters:
+	/// * delegate: The account that the caller would like to remove as a proxy.
+	/// * proxy_type: The permissions currently enabled for the removed proxy account.
+	/// * delay: The announcement period required of the initial proxy. Will generally be zero.
+	fn remove_proxy(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		read_args!(handle, { delegate: Address, proxy_type: u8, delay: u32 });
+
+		let delegate = Runtime::AddressMapping::into_account_id(delegate.into());
+		let proxy_type = Runtime::ProxyType::decode(&mut proxy_type.to_le_bytes().as_slice())
+			.map_err(|_| {
+				RevertReason::custom("Failed decoding value to ProxyType").in_field("proxyType")
+			})?;
+		let delay = delay.into();
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::remove_proxy {
+			delegate,
+			proxy_type,
+			delay,
+		}
+		.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Unregister all proxy accounts for the sender.
+	/// The dispatch origin for this call must be Signed.
+	/// WARNING: This may be called on accounts created by anonymous, however if done, then the
+	/// unreserved fees will be inaccessible. All access to this account will be lost.
+	fn remove_proxies(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::remove_proxies {}.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Checks if the caller has an account proxied with a given proxy type
+	///
+	/// Parameters:
+	/// * delegate: The account that the caller has maybe proxied
+	/// * proxyType: The permissions allowed for the proxy
+	/// * delay: The announcement period required of the initial proxy. Will generally be zero.
+	fn is_proxy(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		read_args!(handle, { delegate: Address, proxy_type: u8, delay: u32 });
+
+		let delegate = Runtime::AddressMapping::into_account_id(delegate.into());
+		let proxy_type = Runtime::ProxyType::decode(&mut proxy_type.to_le_bytes().as_slice())
+			.map_err(|_| {
+				RevertReason::custom("Failed decoding value to ProxyType").in_field("proxyType")
+			})?;
+		let delay = delay.into();
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let is_proxy = ProxyPallet::<Runtime>::proxies(origin)
+			.0
+			.iter()
+			.any(|pd| pd.delegate == delegate && pd.proxy_type == proxy_type && pd.delay == delay);
```

Comment:
Now we are obliged to specify a delay even if we don't want to. We must be able to check if a proxy exists with or without a delay.
The cleanest way is probably to make 2 functions `is_proxy` and `is_proxy_with_delay`.

Reply:
As per the docs https://wiki.polkadot.network/docs/learn-proxies#time-delayed-proxies delay seems to be an integral parameter for a proxy since it's when the announcement would be made before `N` blocks. Even remove proxies would only remove if the `delay` matches. I think this is consistent with the other functions. The `proxyType` and `delay` are both important parameters for equality and as such must be provided.

Reply:
ok